tests/testthat/test-sagemaker.R

```diff
@@ -0,0 +1,204 @@
+        vetiver_sm_model(image_uri = image_uri, role = role, tags = "potato")
+        vetiver_sm_model(image_uri = image_uri, role = role, tags = list("potato"))
+        vetiver_sm_model(image_uri = image_uri, role = role,
+                         vpc_config = list("potato"))
+        vetiver_sm_model(image_uri = image_uri, role = role,
+                         vpc_config = list(Subnets = list(1:3), SecurityGroupIds = 1:3))
+        vetiver_sm_model(image_uri = image_uri, role = role,
+                         vpc_config = list(Subnets = 1:3, SecurityGroupIds = list(1:3)))
+    })
+
+    out <-
+        vetiver_sm_model(
+            image_uri = image_uri,
+            model_name = model_name,
+            role = role,
+            tags = list("new-tag" = "amazing")
+        )
+
+    expect_equal(out, model_name)
+})
+
+test_that("can create SageMaker Endpoint", {
+    mockery::stub(vetiver_sm_endpoint, "sagemaker_client$create_endpoint_config", list())
+    mockery::stub(vetiver_sm_endpoint, "sm_create_endpoint", "example-endpoint-name")
+    model_name <- "vetiver-sagemaker-example_model"
+    instance_type <- "ml.t2.medium"
+
+    expect_snapshot(error = TRUE, {
+        vetiver_sm_endpoint(model_name = model_name, instance_type = instance_type, tags = "potato")
+        vetiver_sm_endpoint(model_name = model_name, instance_type = instance_type, tags = list("potato"))
+    })
+
+    out <-
+        vetiver_sm_endpoint(
+            model_name = model_name,
+            instance_type = instance_type,
+            tags = list("new-tag" = "amazing")
+        )
+
+    expect_equal(out, vetiver_endpoint_sagemaker(model_name))
+})
+
+test_that("can call sm_create_endpoint", {
+    mockery::stub(sm_create_endpoint, "client$create_endpoint", list())
+
+    endpoint_name <- "vetiver-sagemaker-example-model"
+    config_name <- "vetiver-sagemaker-config"
+    client <- paws.machine.learning::sagemaker(list())
+
+    expect_snapshot(
+        out <- sm_create_endpoint(client, endpoint_name, config_name, wait = FALSE)
+    )
+
+    expect_equal(out, endpoint_name)
+})
+
+test_that("can delete SageMaker endpoints", {
+    mock_delete_endpoint_config <- mockery::mock(abort("No config!"), list())
+    mock_delete_model <- mockery::mock()
+    mockery::stub(vetiver_sm_delete, "sagemaker_client$describe_endpoint", list(EndpointConfigName = "the-name"))
+    mockery::stub(vetiver_sm_delete, "sagemaker_client$delete_endpoint_config", mock_delete_endpoint_config)
+    mockery::stub(vetiver_sm_delete, "sagemaker_client$delete_endpoint", list())
+    mockery::stub(vetiver_sm_delete, "sagemaker_client$delete_model", mock_delete_model)
+
+    object <- vetiver_endpoint_sagemaker("vetiver-sagemaker-example-model")
+
+    expect_snapshot(vetiver_sm_delete(object))
+    expect_true(vetiver_sm_delete(object))
+    # Check if delete model and endpoint have been called
+    expect_equal(environment(mock_delete_endpoint_config)$call_no, 2)
```

Comment:
Can we use `mockery::expect_called()` here and below instead of manually getting the env?

Reply:
You may already be familiar with this, but this vignette outlines that:
https://cran.rstudio.com/web/packages/mockery/vignettes/mocks-and-testthat.html

Reply:
Will change it now